packages/x-data-grid/src/components/menu/GridMenu.tsx

```diff
@@ -43,10 +41,9 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-const GridMenuRoot = styled(Popper, {
+const GridMenuRoot = styled('div' as unknown as GridBaseSlots['basePopper'], {
```

Comment:
Todo: use a custom component to throw an error if this is rendered without `as={basePopper}`.